src/main/java/io/cryostat/recordings/RecordingArchiveHelper.java

```diff
@@ -220,6 +220,9 @@ public Future<ArchivedRecordingInfo> deleteRecording(String recordingName) {
                     .message(Map.of("recording", archivedRecordingInfo))
                     .build()
                     .send();
+            if (fs.size(parentPath) == 0) {
```

Comment:
https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#size(java.nio.file.Path)

> The size of files that are not [regular](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#isRegularFile(java.nio.file.Path,%20java.nio.file.LinkOption...)) files is implementation specific and therefore unspecified.

I don't think `.size()` is the best check here due to the above description, since a directory is not a "regular file".

Perhaps `fs.listDirectoryChildren(parentPath).isEmpty()` would do the job?

Reply:
Ah, yes that makes sense thanks I'll try that.

Reply:
There is a test failure because I use fs.deleteIfExists(path) 3 times instead of 2 due to the addition of the method in the code. Is this a problem?

Reply:
Not a problem, sounds like an expected change in behaviour. You'll have to update the unit tests as well.